src/ansys/mapdl/core/_commands/prep7/areas.py

```diff
@@ -0,0 +1,1458 @@
+
+        Reorients areas so that their normals are consistent with that of a specified area.
+
+        If any of the areas have inner loops, the :ref:`anorm` command will consider the inner loops when it
+        reorients the area normals.
+
+        You cannot use the :ref:`anorm` command to change the normal direction of any element that has a
+        body or surface load. We recommend that you apply all of your loads only after ensuring that the
+        element normal directions are acceptable.
+
+        Real constants (such as nonuniform shell thickness and tapered beam constants) may be invalidated by
+        an element reversal.
+
+        See `Revising YourModel
+        <https://ansyshelp.ansys.com/Views/Secured/corp/v232/en/ans_mod/Hlp_G_MOD8_6.html>`_   of the
+        `Modeling and Meshing Guide
+        <https://ansyshelp.ansys.com/Views/Secured/corp/v232/en/ans_mod/Hlp_G_MOD14.html>`_ for more
+        information.
+        """
+        command = f"ANORM,{anum},{noeflip}"
+        return self.run(command, **kwargs)
+
+    def asub(
+        self,
+        na1: str = "",
+        p1: str = "",
+        p2: str = "",
+        p3: str = "",
+        p4: str = "",
+        **kwargs,
+    ):
+        r"""Generates an area using the shape of an existing area.
+
+        Mechanical APDL Command: `ASUB <https://ansyshelp.ansys.com/Views/Secured/corp/v232/en//ans_cmd/Hlp_C_ASUB.html>`_
+
+        Parameters
+        ----------
+        na1 : str
+            Existing area number whose shape is to be used. If ``P1`` = P, graphical picking is enabled and
+            all remaining arguments are ignored (valid only in the GUI).
+
+        p1 : str
+            Keypoint defining starting corner of area.
+
+        p2 : str
+            Keypoint defining second corner of area.
+
+        p3 : str
+            Keypoint defining third corner of area.
+
+        p4 : str
+            Keypoint defining fourth corner of area (defaults to ``P3`` ).
+
+        Notes
+        -----
+
+        .. _ASUB_notes:
+
+        The new area will overlay the old area. Often used when the area to be subdivided consists of a
+        complex shape that was not generated in a single coordinate system. Keypoints and any corresponding
+        lines must lie on the existing area. Missing lines are generated to lie on the given area. The
+        active coordinate system is ignored.
+        """
+        command = f"ASUB,{na1},{p1},{p2},{p3},{p4}"
+        return self.run(command, **kwargs)
+
+    def a(
+        self,
+        p1: str = "",
+        p2: str = "",
+        p3: str = "",
+        p4: str = "",
+        p5: str = "",
+        p6: str = "",
+        p7: str = "",
+        p8: str = "",
+        p9: str = "",
+        p10: str = "",
+        p11: str = "",
+        p12: str = "",
+        p13: str = "",
+        p14: str = "",
+        p15: str = "",
+        p16: str = "",
+        p17: str = "",
+        p18: str = "",
+        **kwargs,
+    ):
```

Comment:
this is missing the typehint for the integer return

Reply:
I agree. I guess we could use decorators for this, and add them to the `mapdl_extended.py`